src/Spatial/Euclidean/Quaternion.cs

```diff
@@ -25,53 +24,62 @@ public Quaternion(double real, double imagX, double imagY, double imagZ)
             _y = imagY;
             _z = imagZ;
             _w = real;
-            _norm = ToNorm(real, imagX, imagY, imagZ);
-            _abs = Math.Sqrt(_norm);
-            _arg = Math.Acos(real/_abs);
+            _normSquared = ToNorm(real, imagX, imagY, imagZ);
+            _norm = Math.Sqrt(_normSquared); //TODO : robust _norm calculating
+            _arg = Math.Acos(real / _normSquared);
```

Comment:
I would just recompute it on demand, and accepting that in very few cases it may be calculated twice.